### ISSUE TITLE
BMT - add custom metrics to track BMT failure items

### ISSUE THREAD

**Issue by va-albers** [department-of-veterans-affairs/va.gov-team#94325]
## Value Statement
**_As an_ OCTO principal with responsibilities to report metrics on application performance,
**_I want to_** be aware of _all_ cases where Benefits systems result in a silent failure, or require that a veteran is notified of a system failure.
**_So that_** we can focus resources on addressing issue, ensure that teams are responding to silent failure cases by having a metric available in Datadog, built in a way that minimizes complexity for the implementing team
- *silent failure* is defined as any case where a veteran is not aware that a request made by them was not completed. This includes but not limited to sidekiq exhaustion, evidence upload failures, Light House status change to failed after pooling, LH state Pending for too long (wait period TBD), or any other failure where veteran received a success status but we failed to deliver its input to the system of record.   
---
### New Feature
1. In all cases where a BMT system operations result in a _silent failure_ write out a Datadog metric which captures the issue 
```
statsd.increment('silent_failure', [tags=[service:"appeal-application",function:"lighthouse upload"])
```
  - the _service_ tag value should be the application's name from the Datadog Service Catalog list of services
  - the _function_ tag value should be selected by the team and clearly represent the application function that had the silent failure.  This should be the original function that failed (such as "lighthouse evidence upload" or "appeal submission", not a failed notification step ("VANotify")
  - we anticipate there are at least three cases where this scenario would happen:
    - the system has a silent failure issue and hasn't been updated to send email to notify the Veteran of the issue.
    - the system does not have an email for veteran at this time (i.e. un-authenticated forms, optional email field, no email in VA profile).
    - FUTURE (after we implement webhook integration into VA Notify) the system had a silent failure, attempted to contact the Veteran via email, but the email delivery failed.

2.  In all cases where a BMT system operation _would have_ resulted in a silent failure, but that silent failure was avoided by notifying the Veteran of the issue by sending an email, write out a Datadog metric which captures the avoided issue 
```
statsd.increment('silent_failure_avoided', [tags=[service:"appeal-application",function:"lighthouse upload"])
```
- The _service_ and _function_ tags should follow the guidelines listed above.

3. If there are cases where we cannot write these metric note these cases with the ticket creator & product owner.  An example of this would be a case where the team only received notification of errors in email, or where an API used by the system does not provide a success/failure response that matches this goal.

## Outcome, Success Measure, KPI(S), and Tracking Link
- Add here

## Design
- Add here

### Enablement team (if needed)
@va-albers 

## Engineering 
- Add here

## Out of scope
- Monitors will be created by VA Enablement team
- Making changes once webhook is in place will be defined and handled by another ticket. Sending email assumes email is success at this point. 

## Open questions
- Add here

## Tasks
- [ ] _Task_
- [ ] _Task_
- [ ] _Task_

## Definition of Done
- [ ] Meets acceptance criteria
- [ ] Passed E2E testing (90% coverage)
- [ ] Passed unit testing (90% coverage)
- [ ] Passed integration testing (if applicable)
- [ ] Code reviewed (internal)
- [ ] Submitted to staging
- [ ] Team approved production verification process
- [ ] Design performs design QA and verifies the implementation matches the design spec
- [ ] Accessibility specialist performs accessibility review (in code or design)
- [ ] Engineering identifies staging users required to test and shares account and credentials with design and product
- [ ] Product performs functional QA and verifies acceptance criteria was met

## Acceptance Criteria
- [ ] _AC_
- [ ] _AC_
- [ ] _AC_

**Comment by va-albers** [comment]
Noting that the code above is pseudocode/code for vets-api should look more like this:
```
tags = ['service:veteran-facing-forms', 'function: form submission to Lighthouse']
StatsD.increment('silent_failure', tags:)
```
